submissions/ArthurGorbenko/Friend's app/js/script.js

```diff
@@ -0,0 +1,282 @@
+};
+
+const makeDarkBackground = () => {
+  CONSTS.BODY.classList.toggle("bodyDark");
+  CONSTS.MENU.classList.toggle("menuDark");
+  CONSTS.DIRECT_MESSAGE_FORM.classList.toggle("direct-messageDark");
+};
+
+const changeColorCards = (cardsClass, buttonsClass) => {
+  let cards = document.getElementsByClassName(cardsClass);
+  cards = [...cards];
+  cards.forEach(card => card.classList.toggle("card-dark"));
+  let buttons = document.getElementsByClassName(buttonsClass);
+  buttons = [...buttons];
+  buttons.forEach(button => button.classList.toggle("button-dark"));
+};
+
+const createNewList = (people, darkId, cardsClass, buttonsClass) => {
+  CONSTS.LIST_OF_CARDS.innerHTML = "";
+  people.forEach(person => createCard(person));
+  if (document.getElementById(DARK_INPUT_ID).checked) {
+    let cards = document.getElementsByClassName(cardsClass);
+    cards = [...cards];
+    cards.forEach(card => card.classList.add("card-dark"));
+    let buttons = document.getElementsByClassName(buttonsClass);
+    buttons = [...buttons];
+    buttons.forEach(button => button.classList.add("button-dark"));
+  }
+  CONSTS.LIST_OF_CARDS.addEventListener("click", writeLetter);
+};
+
+const searchPeople = subString => {
+  if (!subString || subString === " ") {
+    return null;
+  }
+  if (!filteredByGender) {
+    users = usersConst;
+  }
+  subString = subString.toLowerCase();
+  let usersFound = users.filter(
+    user => user.name.first.match(subString) || user.name.last.match(subString)
+  );
+  createNewList(usersFound, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+};
+
+let query = "";
+
+const showGender = genderType => {
+  if (filteredByGender) {
+    users = usersConst;
+  }
+  users = users.filter(user => user.gender === genderType);
+  createNewList(users, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+  filteredByGender = true;
+  if (query) {
+    searchPeople(query);
+  }
+};
+
+const genderFilter = ({ target }) => {
+  if (target.checked) {
+    showGender(target.dataset.gender);
+  }
+};
+
+const sortingProcess = typeOfSort => {
+  const runSort = (a, b) => a.dob.age - b.dob.age;
+  if (typeOfSort === SORT_BY_ASCENDING) {
+    users.sort(runSort);
+  } else {
+    users.sort((a, b) => runSort(b, a));
+  }
+  createNewList(users, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+  searchPeople(query);
+};
+
+const ageSort = ({ target }) => {
+  if (target.checked) {
+    let sortType = target.dataset.render;
+    sortingProcess(sortType);
+  }
+};
+
+const resetRadio = allButtons => {
+  let buttons = [...allButtons];
+  buttons.forEach(button => (button.checked = false));
+};
+
+const resetAllOptions = () => {
+  CONSTS.SEARCH_FIELD.value = "";
+  query = "";
+  resetRadio(radioButtons);
+  createNewList(usersConst, DARK_INPUT_ID, CARD_CLASS, BUTTON_CLASS);
+};
+
+const switchTheme = ({ target }) => {
+  if (target.checked) {
+    makeDarkBackground();
+    changeColorCards("card", "button");
+  }
+};
+
+const addHamburgerOnMobile = () => {
```

Comment:
This will be better to make with css. Hide and show elements in accordance to screen width with help of media query.

Reply:
Yep,rebuilded.